app/models/user.rb

```diff
@@ -6,4 +6,39 @@ class User < ApplicationRecord
 
   has_many :dragons
   has_many :resources
+
+  def can_afford?(dragon_type)
+    resources_list = DragonCost.resources_amount(dragon_type)
+    resources_list.each do |resource|
+      user_amount = resources.find_by(resource_type: resource.resource_type.id)
+      if !resource?(user_amount) || !good_amount?(user_amount, resource)
+        return false
+      end
+    end
+    true
+  end
+
+  def missing_resources_for(dragon_type)
+    missing_resources = 'Lacking resources: '
+    resources_list = DragonCost.resources_amount(dragon_type)
```

Comment:
`DragonCost.resources_amount(dragon_type)` - tutaj bym to trochę odwrócił
zamiast przekazywać do klasy `DragonCost` obiekt `dragon_type` można umieścić metodę w klasie `DragonType` (ale metodę do wykonania na pojedynczej instancji, nie klasową/statyczną):

czyli zamiast
`DragonCost.resources_amount(dragon_type)`
byłoby
`dragon_type.resources_amount`

wydaje mi się to mniej "na około"